client/src/modules/stock/inventory-adjustment/inventory-adjustment.js

```diff
@@ -0,0 +1,273 @@
+      .filter(item => item.lot && item.lot.uuid)
+      .map(item => item.lot.uuid);
+  }
+
+  // add items
+  function addItems(n) {
+    vm.Stock.addItems(n);
+    checkValidity();
+  }
+
+  // remove item
+  function removeItem(item) {
+    vm.Stock.removeItem(item.id);
+    checkValidity();
+  }
+
+  // configure item
+  function configureItem(item) {
+    item._initialised = true;
+    // get lots
+    Stock.lots.read(null, { depot_uuid : vm.depot.uuid, inventory_uuid : item.inventory.inventory_uuid })
+      .then((lots) => {
+        item.lots = lots.filter(lot => !vm.selectedLots.includes(lot.uuid));
+      })
+      .catch(Notify.handleError);
+  }
+
+  function setupStock() {
+    vm.selectedLots = [];
+    vm.Stock.setup();
+    vm.Stock.store.clear();
+  }
+
+  function startup() {
+    vm.movement = {
+      date : new Date(),
+      entity : {},
+    };
+  }
+
+  // ============================ Inventories ==========================
+  function loadInventories(depot) {
+    setupStock();
+
+    Stock.inventories.read(null, { depot_uuid : depot.uuid })
+      .then((inventories) => {
+        vm.selectableInventories = angular.copy(inventories);
+        return Stock.lots.read(null, {
+          depot_uuid : depot.uuid,
+          includeEmptyLot : 0,
+          dateTo : vm.movement.date,
+        });
+      })
+      .then(lots => {
+        addItems(lots.length);
+        vm.Stock.store.data.forEach((item, index) => {
+          const lot = lots[index];
+          item.inventory = {
+            inventory_uuid : lot.inventory_uuid,
+            text : lot.text,
+            code : lot.code,
+          };
+          item.lot = lot;
+          item.quantity = lot.quantity;
+          // added for sorting on these columns
+          item.code = lot.code;
+          item.description = lot.text;
+          item.label = lot.label;
+          // set item lots details
+          configureItem(item);
+        });
+        checkValidity();
+      })
+      .catch(Notify.handleError);
+  }
+
+  // check validity
+  function checkValidity() {
+    const lotsExists = vm.Stock.store.data.every((item) => {
+      return item.quantity >= 0 && item.lot && item.lot.uuid;
+    });
+
+    vm.validForSubmit = (lotsExists && vm.Stock.store.data.length);
+  }
+
+  // ================================= Submit ================================
+  function submit() {
+    // check stock validity
+    checkValidity();
+
+    if (!vm.validForSubmit) { return 0; }
+
+    if (vm.Stock.hasDuplicatedLots()) {
+      return Notify.danger('ERRORS.ER_DUPLICATED_LOT', 20000);
+    }
+
+    const movement = {
+      depot_uuid : vm.depot.uuid,
+      date : vm.movement.date,
+      description : vm.movement.description,
+      is_exit : 0,
+      flux_id : bhConstants.flux.INVENTORY_ADJUSTMENT,
+      user_id : Session.user.id,
+    };
+
+    const lots = vm.Stock.store.data.map((row) => {
+      const out = row.lot;
+      out.oldQuantity = angular.copy(out.quantity);
```

Comment:
Is this just a number?  If so, then you don't need `angular.copy()` - it isn't an object that needs to be de-referenced.